src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -115,16 +113,180 @@ public CodeGenIL(IDictionary<string, IList> input, string filePath)
 
         private Type EmitCoercionCode(Type argType, ParameterInfo param)
         {
+            if (param.ParameterType.IsAssignableFrom(argType)) return argType;
+
             if(argType == typeof(double) && param.ParameterType == typeof(long))
             {
                 EmitOpCode(OpCodes.Conv_I8);
                 return typeof(long);
             }
+            if (argType == typeof(double) || argType == typeof(long) && param.ParameterType == typeof(int))
+            {
+                EmitOpCode(OpCodes.Conv_I4);
+                return typeof(int);
+            }
+            if(argType == typeof(double[]) && typeof(IEnumerable<int>).IsAssignableFrom(param.ParameterType))
```

Comment:
I think you can cover more cases with `IsAssignableFrom` for the `argType` check
`typeof(System.Collections.Generic.IEnumerable<int/double/long/etc>).IsAssignableFrom(argType.GetType())` instead of specifically checking for array of type.

Reply:
But I guess you would need to support convertion of IEnumerables
Ex. `EmitIEnumerableCoercion<Src, Trg>()`